proctor-common/src/test/java/com/indeed/proctor/common/TestTestChooser.java

```diff
@@ -250,4 +404,33 @@ private static TestChooser.Result choose(final ForceGroupsOptions forceGroupsOpt
                 forceGroupsOptions
         );
     }
+
+    private static boolean comparePayloadMap(final Map<String, Object> a, final Map<String, Object> b) {
```

Comment:
Should this be added to the Payload class equals function? Currently it is doing 
`Objects.equals(map, payload.map);` 
This will return the incorrect value when the object within the map is an array.

Reply:
I think it's a great improvement if you update the Payload class equals (assuming nothing depends on the wrong behavior)

Reply:
While testing I noticed that serializing/deserializing does depend on the wrong behavior. Since deserializing will turn Collections such as a normal array to ArrayList. Some options to resolve this would be after deserializing turning any ArrayList into an Array or in the equals to allow comparing ArrayList and arrays.